crates/uv/tests/it/pip_install.rs

```diff
@@ -13145,3 +13145,53 @@ fn install_with_system_interpreter() {
     "
     );
 }
+
+/// Test that a missing Python version is not installed when not using `--target` or `--prefix`.
+#[test]
+fn install_missing_python_no_target() {
+    // Create a context that only has Python 3.11 available.
+    let context = TestContext::new("3.11");
+
+    // Request Python 3.12; which should fail
+    uv_snapshot!(context.filters(), context.pip_install()
+        .arg("--python").arg("3.12")
+        .arg("anyio"), @r###"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    error: No virtual environment found for Python 3.12; run `uv venv` to create an environment, or pass `--system` to install into a non-virtual environment
+    "###
+    );
+}
+
+#[test]
+fn install_missing_python_with_target() {
+    // Create a context that only has Python 3.11 available.
+    let context = TestContext::new("3.11");
+
+    let target_dir = context.temp_dir.child("target-dir");
+
+    // Request Python 3.12 which is not installed in this context. Remove the `UV_PYTHON_DOWNLOADS` value that's set in the default
+    // test context to allow the new version to be installed.
+    uv_snapshot!(context.filters(), context.pip_install()
+        .arg("anyio")
+        .arg("--python").arg("3.12")
+        .arg("--target").arg(target_dir.path())
+        .env_remove(EnvVars::UV_PYTHON_DOWNLOADS), @r###"
```

Comment:
This should instead use https://github.com/astral-sh/uv/blob/257243fec5fc1d85817984e7b1c906b6a9ae3e08/crates/uv/tests/it/common/mod.rs#L514